### PR TITLE
Projects page functionality

### LINKED ISSUE #212: Projects page functionality
Add functionality to the projects page.

### PR DESCRIPTION
**Pull request by Mikxox**
Adding actual functionality to the projects page
closes #212 

~~NOTE: select students page is present but does not work correctly at all!~~
~~NOTE: there is a bug when refresh token is timed out due to wrong hook usage in error parser, this will be fixed soon~~
~~NOTE: deleting project wont work until #233 is merged~~
NOTE: due to backend issues, deleting a project and re-posting a status suggestion don't work

~~Waiting for #210 to be fixed to add pagination~~

~~Waiting for editions to get fixed so I can merge and refactor all the request logic~~

Search currently works by either pressing enter or clicking the search icon, search while typing is not implemented, if we want this we should decide on if we want to launch a new search request every keypress or only every x-seconds or ..

~~Unsure if all this functionality will be part of this pr or split over multiple pr's~~
- [x] Can load projects info from backend
- [x] Can load students for sidebar from backend
- [x] Filter controls on sidebar for students work
- [x] pagination on sidebar works ~~-> waiting for backend~~
- [x] pagination on projects works ~~-> waiting for backend~~
- [x] student sidebar scroll is separate from projects scroll
- [x] Can add a student to a project via drag n drop
- [x] Popup to select role and give reason for adding student
- [x] Popup to edit a project ~~-> waiting for patch endpoint fix~~
- [x] Can delete a project
- [x] Popup to create a new project 
- [x] Popup styling
- [x] ~~Drag n drop styling~~ too much work to implement extra draglayer, fixed somewhat by changing parents
- [x] Request error handling
- [x] Can remove a student from a project
- [x] Projects frontend view updates when added / removed -> without resetting scroll / pagination
- [x] Students sidebar frontend view updates when added / removed -> without resetting scroll / pagination; ~~waiting for 'filter by already assigned' to implement this~~
- [x] hide admin only buttons
- [x] Can search projects by name
- [x] Can search students by name
- ~~[ ] Keep sidebar state when switching between projects & students~~ outside scope of this pr
- [x] correct use of editions ~~-> waiting on editions frontend pr~~

This is branched from #199 projects-page since the layout isn't merged into development yet

~~All of this will need to be refactored once the serializers & edition stuff is merged because currently every get requests returns all the needed information, this will be changed to only returning url's that will all need separate fetch requests~~

If you have trouble with yarn run build; yarn start
go into next.config.js and add the following inside module.exports
```
  typescript: {
    ignoreBuildErrors: true,
  }
```

### CODE REVIEW

**Review by NinjaJay — COMMENTED** [review]
This looks a bit weird
image

File: frontend/pages/[editionName]/students.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    diploma: 'No diploma, I am self taught',
+    diplomaYears: '0',
+    diplomaYear: '0',
+    school: 'self taught',
+    role: ['Video editor', 'Photographer'],
+    bestSkill: 'Marketer',
+    participated: true,
+    studentCoach: true,
+  },
+};
```
Comment (NinjaJay) [inline frontend/pages/[editionName]/students.tsx:132]: Is this still needed?

Comment (Mikxox) [reply frontend/pages/[editionName]/students.tsx]: removed 👍

**Comment by Mikxox** [comment]
> This looks a bit weird image

fixed now 👍

**Review by TomAlard — APPROVED** [review]

**Pull request merged**